apps/src/aichat/views/modelCustomization/CompareModelsDialog.tsx

```diff
@@ -1,17 +1,29 @@
 import React, {useState} from 'react';
 
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
 import AccessibleDialog from '@cdo/apps/templates/AccessibleDialog';
 import {Heading3} from '@cdo/apps/componentLibrary/typography';
 import Button from '@cdo/apps/componentLibrary/button/Button';
 
 import ModelDescriptionPanel from './ModelDescriptionPanel';
 import styles from './compare-models-dialog.module.scss';
+import {ModelDescription} from '../../types';
 
-const CompareModelsDialog: React.FunctionComponent<{onClose: () => void}> = ({
-  onClose,
-}) => {
-  const [chosenModelLeft, setChosenModelLeft] = useState<string>('llama2');
-  const [chosenModelRight, setChosenModelRight] = useState<string>('mistral');
+const CompareModelsDialog: React.FunctionComponent<{
+  onClose: () => void;
+  availableModels: ModelDescription[];
+}> = ({onClose, availableModels}) => {
+  const selectedModelId = useAppSelector(
```

Comment:
I am noticing that on project template back levels (/allthethings levels 3-5), the selected model is not being updated, even on current level. However, the other aiCustomization properties are. Unsure what the discrepancy is.
Tagging @sanchitmalhotra126 who implemented project template backed levels.

Reply:
Looking at my Network tab, I see in `main.json` that the `selectedModel` is updated so that the value is being saved to S3. But the correct value is not being displayed for some reason. This is happening locally for level 1, so wondering if my sources got in a bad state. Curious if this is repro'd by others.

Reply:
Cassi says she sees persistence between page loads though in description 🤔

Reply:
(I should say, I repro what you're seeing Alice :))

Reply:
Yeah, it's interesting that on some levels, the selected model is persisting. I noticed that on the level that it was persisting, `levelAichatSettings` is `undefined` so that `reconciledAiCustomizations` is assigned `EMPTY_AI_CUSTOMIZATIONS` which includes `selectedModel` and default `visibility` is set to `editable`.

Once I ran `bundle exec rake seed:custom_levels`, then NONE of the levels persisted as expected based on code above. We've updated the `levelProperties` a lot the past few days!

So yeah, this should not be a problem after Cassie's follow-up with levelbuilder. Thanks!

Reply:
Hmm I didn't check the other levels. I will re-look at this for the part 2 level builder changes.

Reply:
ohhhh. I think the reason why mine was working was because I had deleted main.json for my level, and I didn't have any old version without the new property. I'll have to follow up about `bundle exec rake seed:custom_levels`. Wrote it down & resolving this in this PR to pick up in the next.

Reply:
It looks like it does actually work now, since I changed the ??s to ||s